misk/src/main/kotlin/misk/environment/Deployment.kt

```diff
@@ -0,0 +1,49 @@
+package misk.environment
+
+/**
+ * The raw environment name. The set of possibilities is unbounded so this should rarely be used.
+ * Instead, use the Deployment type for behavior based on characteristics of an environment.
+ */
+data class Env(val name: String)
+
+/** Deployment describes the context in which the application is running */
+data class Deployment(
+  /**
+   * The name of this deployment. This is used for debugging and should not be parsed.
+   *
+   * All pods in the same deployment will have this same name.
+   */
+  val name: String,
+
+  /**
+   * Whether the service is running in a production environment, having an SLA or handling customer data.
+   */
+  val isProduction: Boolean = false,
+
+  /**
+   * Whether the service is running in a test environment, either locally or in a CI.
+   */
+  val isTest: Boolean = false,
+
+  /**
+   * Whether the service is running on a local developer machine, including as a Docker image.
+   */
+  val isLocalDevelopment: Boolean = false
```

Comment:
what's the reason for `isLocalDev` vs `isTest`

Reply:
The wiring for running locally seems to differ in enough cases.

Reply:
For one example, Admin Dashboard should be run in Development but not Test environments.

Reply:
why? don't you want to test the admin dashboard works in tests?

Reply:
Fair enough :+1: My initial thoughts were prompted by the current state of not having many tests for admin dashboard